From 85298b6c19f3f34a35dce1ab1d1544a49b2d853c Mon Sep 17 00:00:00 2001
From: James Kosin <jkosin@apache.org>
Date: Sat, 17 Mar 2012 01:59:22 +0000
Subject: [PATCH] OPENNLP-471: refactored to search to fix the problems when I
 took out the Index handling. Thanks to Jim for pointing this out and
 producing a test

git-svn-id: https://svn.apache.org/repos/asf/opennlp/trunk@1301854 13f79535-47bb-0310-9956-ffa450edef68
---
 .../tools/namefind/DictionaryNameFinder.java   | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/namefind/DictionaryNameFinder.java b/opennlp-tools/src/main/java/opennlp/tools/namefind/DictionaryNameFinder.java
index d162654c8..24fc1e365 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/namefind/DictionaryNameFinder.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/namefind/DictionaryNameFinder.java
@@ -52,16 +52,18 @@ public Span[] find(String[] tokenStrings) {
 
       for (int endToken = startToken; endToken < tokenStrings.length; endToken++) {
 
-        tokens = new String[(endToken - startToken + 1)];
-        System.arraycopy(tokenStrings, startToken, tokens, 0, (endToken - startToken + 1));
-
-        StringList tokenList = new StringList(tokens);
-
-        if (mDictionary.contains(tokenList)) {
-          foundName = new Span(startToken, endToken + 1);
+        if ((endToken - startToken + 1) > mDictionary.getMaxTokenCount()) {
+          break;
         }
         else {
-          break;
+          tokens = new String[(endToken - startToken + 1)];
+          System.arraycopy(tokenStrings, startToken, tokens, 0, (endToken - startToken + 1));
+
+          StringList tokenList = new StringList(tokens);
+
+          if (mDictionary.contains(tokenList)) {
+            foundName = new Span(startToken, endToken + 1);
+          }
         }
       }
 
